content/common/navigation/engine/guides.yaml

```
@@ -245,6 +245,8 @@ navigation:
                 path: /luau/functions
               - title: Type Coercion
                 path: /luau/type-coercion
+          - title: Luau Type Checking
+            path: /luau/type-checking.md
```

You don't need the `.md` here 🙂 

```suggestion
            path: /luau/type-checking
```